libraries/provider_hab_service.rb

```diff
@@ -0,0 +1,69 @@
+# Copyright:: Copyright 2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/service"
+
+class Chef
+  class Provider
+    class Service
+      class HabSup < Chef::Provider::Service
+        include HabitatCookbook::Helpers
+
+        use_inline_resources
+
+        provides :hab_service
+
+        action :start do
+          declare_resource(:systemd_unit, "#{service_name}.service") do
+            action :create
+            content lazy { unit_content }
+            notifies :restart, "service[#{service_name}]"
+          end
+
+          declare_resource(:service, service_name) do
+            action :start
+          end
+        end
+
+        action :stop do
+          declare_resource(:service, service_name) do
+            action :stop
+          end
+        end
+
+        def service_name
```

Comment:
rename `short_service_name` or something?  i would expect an accessor like this to simply delegate to the new_resource or to be used consistently but you mix and match this `service_name` with `new_resource.service_name` but they're not interchangable.